crates/astria-account-monitor/src/config.rs

```diff
@@ -0,0 +1,184 @@
+    pub force_stdout: bool,
+
+    /// Disables writing trace data to an opentelemetry endpoint.
+    pub no_otel: bool,
+
+    /// The endpoint which will be listened on for serving prometheus metrics
+    pub metrics_http_listener_addr: String,
+}
+
+#[derive(Debug, Clone)]
+pub struct Asset {
+    /// The asset ID of the sequencer chain to monitor.
+    pub asset: Denom,
+}
+
+impl<'de> Deserialize<'de> for Asset {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let value = String::deserialize(deserializer)?;
+        let asset = value.parse().map_err(serde::de::Error::custom)?;
+
+        Ok(Self {
+            asset,
+        })
+    }
+}
+
+impl Hash for Asset {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.asset.hash(state);
+    }
+}
+
+impl Display for Asset {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.asset)
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct SequencerAccountsToMonitor(Vec<Account>);
+
+impl<'de> Deserialize<'de> for SequencerAccountsToMonitor {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let value = String::deserialize(deserializer)?;
+
+        let accounts: Result<Vec<Account>, _> = value
+            .split(',')
+            .map(str::trim)
+            .filter(|s| !s.is_empty())
+            .map(str::parse)
+            .collect();
+
+        let accounts = accounts.map_err(serde::de::Error::custom)?;
+
+        if accounts.is_empty() {
+            return Err(serde::de::Error::custom("empty account list"));
+        }
+
+        let mut items = HashSet::new();
+        if !accounts.iter().all(|item| items.insert(item)) {
+            return Err(serde::de::Error::custom("duplicate accounts"));
+        }
+
+        Ok(Self(accounts))
+    }
+}
+
+#[expect(clippy::into_iter_without_iter, reason = "iter() is not needed")]
+impl<'a> IntoIterator for &'a SequencerAccountsToMonitor {
+    type IntoIter = std::slice::Iter<'a, Account>;
+    type Item = &'a Account;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.0.iter()
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Account {
+    /// The address of the account to monitor.
+    pub address: Address,
+}
+
+impl Hash for Account {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.address.to_string().hash(state);
+    }
+}
+
+impl From<Address> for Account {
+    fn from(address: Address) -> Self {
+        Self {
+            address,
+        }
+    }
+}
+
+impl FromStr for Account {
+    type Err = eyre::Report;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let address = s
+            .parse()
+            .map_err(|e| eyre::eyre!("failed to parse account address: {e}"))?;
```

Comment:
I think I mentioned this before in an (invalidated?) comment: prefer `wrap_err` over explicitly formatting the error. Although in this case I think it'd be fine to just forward the `<Address as FromStr>::Err` without wrapping this in eyre.